OSDOCS-7495: add backup and restore (#63608)
openshift/openshift-docs: microshift_backup_and_restore/microshift-backup-and-restore.adoc

:_content-type: ASSEMBLY
[id="microshift-backup-and-restore"]
= About backing up and restoring application data
include::_attributes/attributes-microshift.adoc[]
:context: microshift-backup-and-restore

toc::[]

> You can manually backup and restore data from applications or workloads running in a {product-title} cluster. If you are using an OSTree system, backup and restore are automated. If you are using a non-OSTree system, then you can manually back up and restore application data.

Reviewer: The thing is we don't actually focus on the application data. All this backup and restore is only about MicroShift data (i.e. etcd database, certificates, etc.).

If application happens to be an operator which stores the data in CRs - then it works as described.
If that's an application with data in some volumes, then we do not cover that

Reviewer 2: ```suggestion
You can manually backup and restore the {product-title} database. If you are using an OSTree system, backup and restore are automated. If you are using a non-OSTree system, then you can manually back up and restore application data.
```

:_content-type: ASSEMBLY
[id="microshift-backup-and-restore"]
= About backing up and restoring {product-title} data
include::_attributes/attributes-microshift.adoc[]
:context: microshift-backup-and-restore

toc::[]

> You can manually backup and restore the {product-title} database. If you are using an RPM-OSTree system, backup and restore are automated. If you are using a non-OSTree system, then you can manually back up and restore {product-title} data. The {product-title} service and Greenboot health checks must be stopped prior to any backups.

Reviewer: I don't think we used term "RPM-OSTree" anywhere. We referred to the non-ostree system as "regular RPM system" or "RPM system".

However I see that [R4E docs](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/composing_installing_and_managing_rhel_for_edge_images/introducing-rhel-for-edge-images_composing-installing-managing-rhel-for-edge-images) use "`rpm-ostree` image".
So perhaps that's the way it should be in the docs? Just my 2 cents

Author: Funny enough, I had settled on "RPM-OSTree" for our docs, but I just looked it up in OCP Supplementary Style Guide, and I see: "RPM, rpm-ostree, OSTree," so the usage is inconsistent. For the moment, I will leave this because we use it in the Install guide. I opened an issue in our Supplementary Style Guide to use "RPM-OSTree" and can update if needed.

> You can manually backup and restore the {product-title} database. If you are using an RPM-OSTree system, backup and restore are automated. If you are using a non-OSTree system, then you can manually back up and restore {product-title} data. The {product-title} service and Greenboot health checks must be stopped prior to any backups.

Reviewer: Placeholder for "Greenboot health checks must be stopped"

:_content-type: ASSEMBLY
[id="microshift-backup-and-restore"]
= About backing up and restoring {product-title} data
include::_attributes/attributes-microshift.adoc[]
:context: microshift-backup-and-restore

toc::[]

> {product-title} automatically creates a backup on every start. These automatic backups are deleted and replaced with the latest backup each time the system restarts. If you are using an RPM-OSTree system, both backup and restore are automated.

Reviewer: > {product-title} automatically creates a backup on every start. (...) If you are using an RPM-OSTree system, both backup and restore are automated.

On non-ostree system there is no automation related to backing up or restoring the data. It's all manual.

* If you are using an `rpm-ostree` system, restoring data is also automated. Otherwise, you must back up and restore data manually.
* You can manually back up and restore the {product-title} database on all supported systems.
* The {product-title} service must be stopped and Greenboot health checks must be completed prior to creating any backups.

include::modules/microshift-service-stopping.adoc[leveloffset=+1]

include::modules/microshift-backing-up-manually.adoc[leveloffset=+1]

//additional resources for backing-up module
[role="_additional-resources"]
.Additional resources
* xref:../microshift_install/microshift-install-rpm.adoc#microshift-service-stopping_microshift-install-rpm[Stopping the MicroShift service]
* xref:../microshift_install/microshift-install-rpm.adoc#microshift-service-starting_microshift-install-rpm[Starting the MicroShift service]

include::modules/microshift-restoring-data-backups.adoc[leveloffset=+1]

//additional resources for restoring-data module
[role="_additional-resources"]
.Additional resources
* xref:../microshift_install/microshift-install-rpm.adoc#microshift-service-stopping_microshift-install-rpm[Stopping the MicroShift service]

> * xref:../microshift_install/microshift-install-rpm.adoc#microshift-service-starting_microshift-install-rpm[Starting the MicroShift service]

Author: Renaming a series of modules used primarily in other places is out of scope of this PR, but thank you for the suggestion.

> * xref:../microshift_install/microshift-install-rpm.adoc#microshift-service-starting_microshift-install-rpm[Starting the MicroShift service]

Reviewer 3: Sorry for the confusion. I tested those links, and they seemed to be incorrect. I added those suggestions to point to what looked like the correct pages.
